vrpn_Oculus.C

```diff
@@ -0,0 +1,337 @@
+/** @file	vrpn_Oculus.C
+@brief	Drivers for various Oculus devices.  Initially, just the DK2.
+
+  @date 2015
+  @copyright 2015 ReliaSolve.com
+  @author ReliaSolve.com russ@reliasolve.com
+  @license Standard VRPN license.
+*/
+
+// Based on the OSVR hacker dev kit driver by Kevin Godby.
+// Based on Oliver Kreylos' OculusRiftHIDReports.cpp and OculusRift.cpp.
+
+#include "vrpn_Oculus.h"
+
+#include "vrpn_BaseClass.h"  // for ::vrpn_TEXT_NORMAL, etc
+
+VRPN_SUPPRESS_EMPTY_OBJECT_WARNING()
+
+#if defined(VRPN_USE_HID)
+
+// USB vendor and product IDs for the models we support
+static const vrpn_uint16 OCULUS_VENDOR = 0x2833;
+static const vrpn_uint16 DK2_PRODUCT = 0x0021;
+
+vrpn_Oculus_DK2::vrpn_Oculus_DK2(bool enableLEDs,
+  const char *name, vrpn_Connection *c,
+  double keepAliveSeconds)
+    : vrpn_Analog(name, c)
+    , vrpn_HidInterface(m_filter =
```

Comment:
this is not a standard way of specifying things - looks like a named parameter, but I think it's actually doing local initialization and then initializing the base class. (Is that right?) Is that intentional? Any clearer way to do it?